padinfo/padinfo.py

```diff
@@ -512,32 +513,80 @@ async def collabscroll(self, ctx, *, query: str):
             await ctx.send(self.makeFailureMsg(err))
 
 
+    async def _parse_ls_query(self, query : str) -> (str, str): 
+        '''Helper function for leaderskill. Takes in a query 
+        and returns the left or right'''
+        if not query: 
+            raise Exception('Query seems to be empty.')
+
+        # i assume neither '"' nor '/' will be used as part of card search
+        # this handles splitting with quotewrapping and /
+        # also comma splitting if there are quotes
+        if '"' in query or '/' in query or ',' in query: 
+            p = r'(?:(?:([^"/,]*))|(?:"([^"/]*)"))\s*(?:/|,)?\s*(?:(?:(?:([^"/,]*))|(?:"([^"/]*)")))?'
+            m = re.fullmatch(p, query)
+            if m: 
+                match = [ma.strip() for ma in m.groups() if (ma and ma.strip() != '')]
+
+                if len(match) in (1,2): 
+                    return (match[0], match[len(match)-1])
+                
+                #otherwise match is totally empty, ie ^ls "" ""
+                raise Exception('Queries seem to be empty.')
+
+            # incorrectly fails to match if there is are commas with slash as primary separator 
+            if '/' in query and ',' in query: #repeat regex search 
+                pp = r'(?:(?:([^"/]*))|(?:"([^"/]*)"))\s*/\s*(?:(?:(?:([^"/]*))|(?:"([^"/]*)")))?'
+                mm = re.fullmatch(pp, query)
+                if mm: 
+                    mmatch = [ma.strip() for ma in mm.groups() if (ma and ma.strip() != '')]
+                    if len(mmatch) in (1,2): 
+                        return (mmatch[0], mmatch[len(mmatch)-1])
+                    #otherwise match is totally empty, ie ^ls "" ""
+                    raise Exception('Queries seem to be empty.')
+
+            raise Exception('Query format is invalid. Only use 1 separator (slash or comma) \
+or surround arguments in quotes.')
+
+        #space exists, this is multiple words
+        #first check if leading words are prefixes 
+        words = query.split()
+        nm, err, debug_info = await self._findMonster(query)
+        if nm and all(word in nm.prefixes for word in words[:-1]): 
+            return (query, query)
+
+        #leading words are not prefixes, spaces exist. 
+        if len(words) >= 3: 
+            raise Exception('Too many inputs given. Limit to 1 or 2, or use quote wrappers or a separator (/ or ,) \
+to separate.') 
+        
+        #it's 2 words, no prefix
+        return (words[0], words[1])
+
     @commands.command(aliases=['leaders', 'leaderskills', 'ls'])
-    async def leaderskill(self, ctx, left_query: str, right_query: str = None, *, bad=None):
+    async def leaderskill(self, ctx, *, query : str):
```

Comment:
If you instead change this signature to:
```
async def leaderskill(self, ctx, left_query: str, right_query: str = None, *, rest_of_query=None):
```
and then immediately return when rest_of_query is None, you will have for free handled the case when quotes are used. Does this help?

Reply:
Possibly, but that has the issue of the default space splitting; something like ^ls horus/revo horus would be harder with it.

Reply:
ah true, good point

Reply:
ok I think this is still the right signature. Then there's a few cases:

First, assume rest_of_query is None.

1. No delimiters in either `left_query` (lq) or `right_query` (rq). Great! Return.
2. A delimiter in lq but not rq, or the other way around. Then almost certainly the sender actually meant the delimiter to be the delimiter of the query. Let's check is it a comma and is there a space inside of this argument? If so then it's probably an exact name so let's respect it; if not then merge with case that rest_of_query is not None.
3. both have delimiters -> are they both exact names? if not then error.

If rest_of_query is not None then assume we had no quotes whatsoever and split on the delimiter using re.split.